source/agora/consensus/protocol/Nominator.d

```diff
@@ -817,6 +814,13 @@ extern(D):
                 "Error: {}", ex.msg);
             return ValidationLevel.kInvalidValue;
         }
+
+        // Per SCP rules, once we CONFIRM a NOMINATE; we can't
+        // nominate new values. Keep track of the biggest slot_idx we confirmed
+        // a NOMINATE on
+        if (slot_idx > this.done_nominating)
```

Comment:
I don't think this is the right place to do this. It assumes the node will confirm something as soon as its validated and this function returns.

Isn't there a confirm event handler virtual method somewhere in SCP? Or something similar to that.

Reply:
Maybe something from here?

https://github.com/stellar/stellar-core/blob/54b03f755ae5d5aa12a799c8f1ee4d87fc9d1a1d/src/scp/SCPDriver.h#L205-L240

Reply:
Other than `updatedCandidateValue` they look unrelated. It's called a bit further into the process, seems like a better place to do this. 

I moved it there.

Reply:
`emitEnvelope` would probable be even better. We could check for an outgoing CONFIRM envelope.

Reply:
We can squash the last commit once we are happy with it.